app/src/main/java/com/github/swent/echo/viewmodels/association/AssociationViewModel.kt

```diff
@@ -0,0 +1,142 @@
+package com.github.swent.echo.viewmodels.association
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.swent.echo.authentication.AuthenticationService
+import com.github.swent.echo.connectivity.NetworkService
+import com.github.swent.echo.data.model.Association
+import com.github.swent.echo.data.model.Event
+import com.github.swent.echo.data.repository.Repository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+
+enum class AssociationPage(var association: Association) {
```

Comment:
why do you need an empty association for every screen ?

Reply:
I agree it's a bit weird but i didn't find a better solution. It's for DETAILS which need to take an Association as argument to know which association we want to display details

Reply:
Oh okay, I don't know if there is a better solution either 🤔 It's okay for now and we could always change it later